### PR TITLE
Reimplement the add-pr-body interceptor as a ClusterInterceptor

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

This commit adds a cluster interceptor flavor of the add pr body interceptor.
The old webhook interceptor has been left untouched.

BREAKING: For the ClusterInterceptor flavor, the PR url input has to now come
in via the `request.extensions ` field instead of the old `request.body` field.
Similarly, the output pr-body is part of the extensions field instead of being
merged back into the body.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by afrittoli** [comment]
Hi @dibyom, thank you for starting this! Are you still working on it? I could try and pick up your PR if you don't have time to work on it anymore. Thank you!

**Comment by dibyom** [comment]
> Hi @dibyom, thank you for starting this! Are you still working on it? I could try and pick up your PR if you don't have time to work on it anymore. Thank you!

hey @afrittoli almost done, just need to test + update the docs. I'll update this PR in the next couple of days.

**Comment by dibyom** [comment]
> Any update on this one? Anything I can do to help?

Sorry for the major delay. Should be ready for a look now 🙏

File: tekton/ci/interceptors/add-pr-body/README.md [hunk 1]
```diff
@@ -1,10 +1,105 @@
 # Add PR Body
 
-`add-pr-body` is a custom interceptor for Tekton Triggers that enriches the
-payload of an incoming request with the JSON representation of a pull request,
-as returned by the GitHub API.
+This folder contains two implementaions of the same interceptor that enriches the payload of an incoming request with
```
Comment (afrittoli) [inline tekton/ci/interceptors/add-pr-body/README.md:3]: NIT implementaions -> implementations

File: tekton/ci/interceptors/add-pr-body/README.md [hunk 2]
```diff
@@ -1,10 +1,105 @@
...
+  interceptors:
+    - cel:
+        filter: >- // TODO: This can be part of the interceptor itself
+          body.action == 'created' &&
+          in('pull_request', body.issue) &&
+          && body.issue.state == 'open' &&
+          body.comment.body.matches('^/test($| [^ ]*$)')
+        overlays:
+        - key: add-pr-body.pull-request-url
+          expression: "body.issue.pull_request.url"
```
Comment (afrittoli) [inline tekton/ci/interceptors/add-pr-body/README.md:62]: I think the new syntax looks something like:

```yaml
      interceptors:
        - name: "Filter the GitHub org and actions and add git_clone_depth"
          ref:
            name: "cel"
```

Should the `add-pr-body` interceptor be used in this example?

File: tekton/ci/interceptors/add-pr-body/pkg/pr.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
+/*
+ Copyright 2021 The Tekton Authors
```
Comment (afrittoli) [inline tekton/ci/interceptors/add-pr-body/pkg/pr.go:2]: 2022?

File: tekton/ci/interceptors/add-pr-body/cmd/interceptor/main.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 The Tekton Authors
```
Comment (afrittoli) [inline tekton/ci/interceptors/add-pr-body/cmd/interceptor/main.go:2]: 2022

**Review by afrittoli — COMMENTED** [review]
Thanks for this. I a couple of comments on the docs but mostly looks great!
The existing interceptor has nightly builds, and I wonder if we should replace it with the new one instead of having two in the same folder, which will make the nightly builds harder to manage. Alternatively we could have two separate folders... wdyt?

**Comment by dibyom** [comment]
Thanks, I think I'm going to move this to a separate folder then to make it easier.

File: tekton/ci/interceptors/add-pr-body/README.md (hunk 2 above)
Comment (dibyom) [reply tekton/ci/interceptors/add-pr-body/README.md]: yup, it should :)

**Comment by dibyom** [comment]
@afrittoli I updated the PR. I also added a couple of TODOs around making this easier to use. Also, we could potentially merge this code with the packaged GitHub interceptor https://github.com/tektoncd/triggers/issues/1339

**Review by afrittoli — COMMENTED** [review]
Thanks a lot @dibyom for this!
I really like the idea to embed this as opt-in in the cluster interceptor. 

If no-one does that before I will start testing this on dogfooding once I'm back from PTO.
I will try to repeat the process for the other interceptor as well - and for that one too I wonder if that's something we could get into core interceptors as well, as it's a pretty common use case to filter based on the owner of the PR.

**Review by afrittoli — COMMENTED** [review]
/approve

**Pull request merged**